streams/src/main/java/org/apache/kafka/streams/processor/TimestampExtractor.java

```diff
@@ -27,17 +27,19 @@
 public interface TimestampExtractor {
 
     /**
-     * Extracts a timestamp from a record.
+     * Extracts a timestamp from a record. The timestamp must be positive. If a negative timestamp is returned,
```

Comment:
I'd suggest:

> Extracts a timestamp from a record.  The timestamp must be positive to be considered a valid timestamp.  Returning a negative timestamp will cause the record not to be processed.

Question: What happens when `timestamp == 0`?  Is this a valid or an invalid timestamp?  The answer should get included in the javadoc.